### PR TITLE
Improved coverage for array and object test 

### PR DESCRIPTION
**Pull request by frankweate**
closes gavv#184
array_test.go:
Added test cases to IsEqualUnordered(), NotEqualUnordered() , NotConsistsOf(), ContainsOnly(), NotContainsOnly(), Every(), Filter()

object_test.go:
Added test cases to ContainsValue(), Every(), Filter()

Please let me know if there is anything  you wish for me to add or change, Thanks!

### CODE REVIEW

File: array_test.go [hunk 1]
```diff
@@ -666,7 +682,11 @@ func TestArray_IsEqualUnordered(t *testing.T) {
 		value.chain.assertFailed(t)
 		value.chain.clearFailed()
 
-		value.NotEqualUnordered(func() {})
```
Comment (gavv) [inline array_test.go:669]: I guess the block:

```go
		value.NotEqualUnordered(func() {})
		value.chain.assertFailed(t)
		value.chain.clearFailed()
```

was lost accidentally, we should restore it.

File: array_test.go [hunk 2]
```diff
@@ -666,7 +682,11 @@ func TestArray_IsEqualUnordered(t *testing.T) {
...
+		value.IsEqualUnordered("123")
+		value.chain.assertFailed(t)
+		value.chain.clearFailed()
+
+		value.NotEqualUnordered("123")
 		value.chain.assertFailed(t)
 		value.chain.clearFailed()
```
Comment (gavv) [inline array_test.go:691]: Which branches are we covering with these two blocks? Aren't they already covered with:

```go
		value.IsEqualUnordered(func() {})
		value.chain.assertFailed(t)
		value.chain.clearFailed()

		value.NotEqualUnordered(func() {})
		value.chain.assertFailed(t)
		value.chain.clearFailed()
```

?

File: array_test.go [hunk 3]
```diff
@@ -874,6 +894,10 @@ func TestArray_ConsistsOf(t *testing.T) {
 		value.NotConsistsOf(123, "foo")
 		value.chain.assertFailed(t)
 		value.chain.clearFailed()
+
+		value.NotConsistsOf(make(chan int)) // bad type make(chan int)
+		value.chain.assertFailed(t)
+		value.chain.clearFailed()
```
Comment (gavv) [inline array_test.go:900]: Isn't this the same as:

```go
		value.NotConsistsOf(func() {})
		value.chain.assertFailed(t)
		value.chain.clearFailed()
```

which we already have?

File: array_test.go [hunk 4]
```diff
@@ -1229,6 +1253,18 @@ func TestArray_ContainsOnly(t *testing.T) {
...
+
+		value.NotContainsOnly("foo", 123, "bar")
+		value.chain.assertNotFailed(t)
+		value.chain.clearFailed()
+
+		value.NotContainsOnly(make(chan int)) // bad type make(chan int)
+		value.chain.assertFailed(t)
+		value.chain.clearFailed()
```
Comment (gavv) [inline array_test.go:1267]: Isn't this the same as:

```go
		value.NotContainsOnly(func() {})
		value.chain.assertFailed(t)
		value.chain.clearFailed()
```

which we already have?

File: array_test.go [hunk 5]
```diff
@@ -1482,6 +1518,13 @@ func TestArray_Every(t *testing.T) {
 		array.chain.assertNotFailed(t)
 	})
 
+	t.Run("nil func", func(t *testing.T) {
```
Comment (gavv) [inline array_test.go:1521]: Let's rename this to "invalid argument" and move to the end of the test, for consistency with other tests.

File: array_test.go [hunk 6]
```diff
@@ -1660,6 +1703,14 @@ func TestArray_Filter(t *testing.T) {
 		filteredArray.chain.assertNotFailed(t)
 	})
 
+	t.Run("nil func", func(t *testing.T) {
```
Comment (gavv) [inline array_test.go:1706]: Let's rename this to "invalid argument" and move to the end of the test, for consistency with other tests.

File: object_test.go [hunk 7]
```diff
@@ -729,6 +729,10 @@ func TestObject_ContainsValue(t *testing.T) {
 		value.NotContainsValue("XXX")
 		value.chain.assertNotFailed(t)
 		value.chain.clearFailed()
+
+		value.ContainsValue(make(chan int))
+		value.chain.assertFailed(t)
+		value.chain.clearFailed()
```
Comment (gavv) [inline object_test.go:735]: Let's extract this into subtest "invalid argument" and move it to the end of the test.

Let's also add the similar check for NotContainsValue to the same subtest.

File: object_test.go [hunk 8]
```diff
@@ -1433,6 +1444,13 @@ func TestObject_Filter(t *testing.T) {
 		object.chain.assertNotFailed(t)
 	})
 
+	t.Run("nil func", func(t *testing.T) {
```
Comment (gavv) [inline object_test.go:1447]: Let's rename this to "invalid argument" and move to the end of the test.

**Review by gavv — COMMENTED** [review]
Thanks for PR! Here are a few comments.

File: array_test.go (hunk 2 above)
Comment (frankweate) [reply array_test.go]: Removed these 2 cases

File: array_test.go (hunk 1 above)
Comment (frankweate) [reply array_test.go]: added this back

File: array_test.go (hunk 3 above)
Comment (frankweate) [reply array_test.go]: you're right, it follows the same path in the function, so i've removed it

File: array_test.go (hunk 4 above)
Comment (frankweate) [reply array_test.go]: removed it

File: array_test.go (hunk 5 above)
Comment (frankweate) [reply array_test.go]: Changed to the consistency of other tests

File: array_test.go (hunk 6 above)
Comment (frankweate) [reply array_test.go]: Changed format to the consistency of other tests

File: object_test.go (hunk 7 above)
Comment (frankweate) [reply object_test.go]: added this new subtest and the Not containsValue check

File: object_test.go [hunk 9]
```diff
@@ -1196,6 +1200,13 @@ func TestObject_Every(t *testing.T) {
...
```
[the remark this answers is not in the document]
Comment (frankweate) [reply object_test.go]: Changed format to the consistency of other tests

File: object_test.go (hunk 8 above)
Comment (frankweate) [reply object_test.go]: Changed format to the consistency of other tests

**Pull request merged**